From 66a91c35782618063cc2c1e5fc1bc3714f31862c Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sat, 2 Nov 2024 19:52:31 +0100
Subject: [PATCH 10/25] board: solidrun: lx2160acex7: disable second usb on
 lx2162

LX2162 only has single USB controller, disable the second one from
board_fix_fdt.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/lx2160acex7/lx2160a.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/board/solidrun/lx2160acex7/lx2160a.c b/board/solidrun/lx2160acex7/lx2160a.c
index af0a071488a..e0a9e6c51eb 100644
--- a/board/solidrun/lx2160acex7/lx2160a.c
+++ b/board/solidrun/lx2160acex7/lx2160a.c
@@ -79,6 +79,10 @@ int board_fix_fdt(void *fdt)
 		{ "pf_ctrl", "ctrl" }
 	};
 	int off = -1, i = 0;
+	u32 is_lx2162 = get_svr() & 0x800;
+
+	if (is_lx2162)
+		do_fixup_by_path_string(fdt, "/usb3@3110000", "status", "disabled");
 
 	board_fix_fdt_eth(fdt);
 
-- 
2.43.0

